src/vdom/component.js

```diff
@@ -49,6 +49,18 @@ export function setComponentProps(component, props, opts, context, mountAll) {
 	if (component.__ref) component.__ref(component);
 }
 
+export function catchErrorInComponent(error, component) {
+	for (; component; component = component._ancestorComponent) {
+		if (component.componentDidCatch) {
+			try {
```

Comment:
Do we _need_ another try/catch here? I would think that if `componentDidCatch` exists, the error is passed to it & it ends there. If the method does not exist, it's uncaught so the component (tree) is unmounted.

Reply:
Errors that occur in `componentDidCatch` should bubble to ancestors, regardless of what lifecycle method generated the error or from what path that lifecycle method occurred. Error paths _should_ be rare, so it's fine for it to be slow.